cgo/lapack.go

```diff
@@ -228,7 +228,7 @@ func (impl Implementation) Dgeqrf(m, n int, a []float64, lda int, tau, work []fl
 // with native, lwork == -1 will not run Dgeqrf but will instead write the minimum
 // work necessary to work[0]. If len(work) < lwork, Dgeqrf will panic.
 func (impl Implementation) Dgels(trans blas.Transpose, m, n, nrhs int, a []float64, lda int, b []float64, ldb int, work []float64, lwork int) bool {
-	mn := min(m, n)
+	mn := max(m, n)
```

Comment:
It looks to me like the real fix is to change the line below (no comment available there) to `checkMatrix(max(m, n), nrhs, b, ldb)`, and revert this.